query-engine/core/src/interpreter/interpreter_impl.rs

```diff
@@ -276,28 +292,28 @@ impl<'conn> QueryInterpreter<'conn> {
                 then,
                 else_: elze,
             } => Box::pin(async move {
-                self.log_line(level, || "IF");
+                let predicate = func();
+                self.log_line(level, || format!("if <lambda condition> = {predicate} {{"));
 
-                if func() {
+                let result = if predicate {
                     self.interpret(Expression::Sequence { seq: then }, env, level + 1, trace_id)
                         .await
                 } else {
                     self.interpret(Expression::Sequence { seq: elze }, env, level + 1, trace_id)
                         .await
-                }
+                };
+                self.log_line(level, || "}");
+                result
             }),
 
-            Expression::Return { result } => Box::pin(async move {
-                self.log_line(level, || "RETURN");
-                Ok(*result)
-            }),
+            Expression::Return { result } => Box::pin(async move { Ok(*result) }),
```

Comment:
Would you mind keeping a log of return? I know it's just supposed to be a passthrough so logging it doesn't add value in theory, but this thing is not guaranteed to be bug free and removing it in the logs might make someone waste a lot of time if return returns out to be buggy

Reply:
Oops, I actually removed this one by accident when I was cutting down the debug prints. I will bring it back!